### DISCUSSION TITLE
How to set mesh priority of an object to a specific number

### DISCUSSION THREAD

**Discussion by mjmashayekhi** [ansys/pyaedt#4641] [Q&A] [answered]
I need to set the mesh priority of an object to a specific value. I used the command:
ipk.mesh.add_priority(1, ["bbox"], bbox, [5])

However it doesn't set the priority to 5 in this case, even thought it doesn't give any error message too.

**Comment by lorenzovecchietti** [comment, marked as answer]
Hi, unfortunately this method is severely limited by native API.

They allow either to set all the priorities at once or to set just one priority. The way the pya dt function is implemented allows to set just one priority (see bug #4645).

I'll try to implement a better method soon. In the time being please use native APIs (and remember that you have to provide the full list of priority levels with the appropriate objects).

**Reply by lorenzovecchietti** [reply]
This means something like:
```
ipk.modeler.oeditor.UpdatePriorityList(
	[
		"NAME:UpdatePriorityListData",
		[
			"NAME:PriorityListParameters",
			"EntityType:="		, "Object",
			"EntityList:="		, "Box3",
			"PriorityNumber:="	, 2,
			"PriorityListType:="	, "3D"
		],
		[
			"NAME:PriorityListParameters",
			"EntityType:="		, "Object",
			"EntityList:="		, "Box2",
			"PriorityNumber:="	, 3,
			"PriorityListType:="	, "3D"
		]
	])
```
